src/main/java/ch/njol/skript/entity/VillagerData.java

```diff
@@ -49,25 +51,34 @@ public class VillagerData extends EntityData<Villager> {
 		// NORMAL(-1), FARMER(0), LIBRARIAN(1), PRIEST(2), BLACKSMITH(3), BUTCHER(4), NITWIT(5);
 		
 		Variables.yggdrasil.registerSingleClass(Profession.class, "Villager.Profession");
-		
+
+		professions = new ArrayList<>();
 		if (Skript.isRunningMinecraft(1, 14)) {
 			EntityData.register(VillagerData.class, "villager", Villager.class, 0,
 					"villager", "normal", "armorer", "butcher", "cartographer",
 					"cleric", "farmer", "fisherman", "fletcher",
 					"leatherworker", "librarian", "mason", "nitwit",
 					"shepherd", "toolsmith", "weaponsmith");
-			professions = Arrays.asList(Profession.values());
+			professions = Arrays.asList(Profession.NONE, Profession.ARMORER, Profession.BUTCHER, Profession.CARTOGRAPHER,
+					Profession.CLERIC, Profession.FARMER, Profession.FISHERMAN, Profession.FLETCHER, Profession.LEATHERWORKER,
+					Profession.LIBRARIAN, Profession.MASON, Profession.NITWIT, Profession.SHEPHERD, Profession.TOOLSMITH,
+					Profession.WEAPONSMITH);
```

Comment:
My only concern with this change is that it needs to be manually updated (though I suppose the text does too...) is there no way to dynamically base this on what's available from the class?

Reply:
They should all be available from Registry.VILLAGER_PROFESSION

Reply:
I did it this way as the ordering of the elements is important (see matched pattern usage). I do not believe there are any guarantees on the ordering of the registry entries, and I'm not sure if its contents may differ depending on whether any data packs are being used. I could leave a comment in the code explaining this. I would like to come back to this on dev/feature though to rework this class to use registries (would likely require EntityData reworks)

Reply:
a comment would be nice, with a TODO, but that sound good to me

Reply:
Sure, I'll add that. It doesn't really need to be dynamic (at this point) anyways as the lang files and class would already need to be manually updated for new types.